src/click/core.py

```diff
@@ -2352,6 +2352,9 @@ def process_value(self, ctx: Context, value: t.Any) -> t.Any:
             raise MissingParameter(ctx=ctx, param=self)
 
         if self.callback is not None:
+            # The callback is receiving the default value as-is, which let the
+            # developer decide how to handle the different cases of None, UNSET, or any
+            #  other value.
```

Comment:
Make sense. So this exposes to UNSET in the callback? If so, it looks like we will have to expose UNSET as public. Is this [this callback](https://click.palletsprojects.com/en/stable/advanced/#callbacks-for-validationl). @kdeldycke @davidism

Reply:
Just documented the `UNSET` on callback in: https://github.com/pallets/click/pull/3030/commits/e6870a68090926a0c49b79243fef9acf6550809e